starts of a GithubBackend

clux opened the pull request:
Needs a new version of hubcaps that supports release assets before this can be used, but this contains the preliminary type setup. This would implement #1 


Got stuck on having to add logic to hubcaps.
Think I had it almost working in https://github.com/lalbuild/hubcaps/commit/a13f4400b09ca04c83070f80eed105c971033a05

- [ ] need to upgrade hyper to use new hubcaps (about time anyway)
- [ ] fix progressbars for tokio (see [indicatif tokio example](https://github.com/mitsuhiko/indicatif/blob/master/examples/tokio.rs) )
- [ ] actually finish off the backend

bencord0 commented on Cargo.toml:
```diff
@@ -1,9 +1,9 @@
 description = "A strict, language-agnostic build system and dependency manager"
 license = "MIT"
+name = "lal"
+version = "3.7.0"
```
alphabetical keys?

clux replied:
ugh, it's just different tools keep re-ordering them. both clippy and cargo-edit change the key order in different ways. maybe one of them parses into a hashmap rather than a btreemap.

bencord0 commented on src/storage/github.rs:
```diff
@@ -0,0 +1,126 @@
+#[derive(Serialize, Deserialize, Clone)]
+pub struct GithubCredentials {
+    /// Personal access token with upload access
+    pub token: String,
```
Document how to get hold of one of these.

clux commented:
crucial TODO: https://github.com/softprops/hubcaps/blob/28f076966fbdab030aa16158ff0b987fadaaa3c9/src/releases/mod.rs#L29

clux committed:
starts of a GithubBackend

hubcaps is missing the release assets api unfortunately though.
So we can create a release, but we can't attach our artifact to it yet.

Stashing current work in a branch.

clux commented:
K. Legit failure. Finally travis.
